theano/tensor/nnet/abstract_conv.py

```diff
@@ -1723,12 +1730,30 @@ def perform(self, node, inp, out_):
         if not self.filter_flip:
             kern = kern[(slice(None),) * (kern.ndim - self.convdim) + (slice(None, None, -1),) * self.convdim]
 
-        # from (nFilters, out_rows, out_cols, nChannels, kH, kW)
-        # to (nFilters, nChannels, out_rows, out_cols, kH, kW)
         if self.unshared:
+            out_shape = get_conv_output_shape(img.shape, kern.shape,
+                                              mode, self.subsample, self.filter_dilation)
+            if kern.shape[1:1 + self.convdim] != out_shape[2:2 + self.convdim]:
+                raise ValueError('Kernel shape {} does not match '
+                                 'output size {}'.format(kern.shape[1:1 + self.convdim],
```

Comment:
Maybe make that computed output size, so it's clear that it's not an user-defined input that has the wrong shape?